guides/plugins/plugins/storefront/add-custom-controller.md

```diff
@@ -1,2 +1,215 @@
+
+{% code title="<plugin root>/src/Storefront/Controller/ExampleController.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+}
+```
+{% endcode %}
+
+Now we can create a new example method with a `Route` annotation which has to contain our route, in this case it will be `/example`.
+The route defines how our new method will be accessible.
+
+Below you can find an example implementation of a controller method including a route, where we render an `example.html.twig` template file with a template variable `example`.
+
+{% code title="<plugin root>/src/Storefront/Controller/ExampleController.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+    /**
+    * @Route("/example", name="frontend.example.example", methods={"GET"})
+    */
+    public function showExample(): Response
+    {
+        return $this->renderStorefront('@SwagBasicExample/storefront/page/example/index.html.twig', [
+            'example' => 'Hello world'
+        ]);
+    }
+}
+```
+{% endcode %}
+
+The name of the method does not really matter, but it should somehow fit its purpose.
+More important is the `Route` annotation, that points to the route `/example`.
+Inside the method, we're using the method `renderStorefront` to render a twig template file in addition with
+the template variable `example`, which contains `Hello world`. This template variable will be usable in the rendered
+template file.
+The method `renderStorefront` then returns a `Response`, as every routed controller method has to.
+
+
+It is also possible to define the `RouteScope` per route and also multiple scopes at once, as you can see below.
+
+{% code title="<plugin root>/src/Storefront/Controller/ExampleController.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Storefront\Controller;
+
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Shopware\Storefront\Controller\StorefrontController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @RouteScope(scopes={"storefront"})
+ */
+class ExampleController extends StorefrontController
+{
+    /**
+    * @RouteScope(scopes={"storefront", "custom-scope"})
```

Comment:
I won't bring the custom example here. otherwise you need to explain what's its purpose and that would be worth an own article :grin: